qiime2/core/path.py

```diff
@@ -24,6 +24,12 @@ def _party_parrot(self, *args):
 
 class OwnedPath(_ConcretePath):
     def __new__(cls, *args, **kwargs):
+        from qiime2.core.cache import get_cache
+
+        cache = get_cache()
+        kwargs['prefix'] = os.path.join(cache.process_pool.path,
+                                        'q2-%s-' % cls.__name__)
```

Comment:
Let's put this under a `tmp/` directory in the process pool (creating if needed). Maybe there should be a helper on the cache itself to return the temp directory?